pkg/cmd/pulumi/packagecmd/project_sdk_link.go

```diff
@@ -915,6 +915,18 @@ func ProviderFromSource(
 		return p, specOverride, nil
 	}
 
+	if pluginSpec.Version != nil {
```

Comment:
I'm surprised to see this fix here, rather than in the registry resolution code.  The bug only affects that, and the rest of the package resolution code works correctly already.  

Would you mind adding some PR description on *why* this is the right fix?